File: sharedse/src/test.disabled/java/com/ustadmobile/port/sharedse/impl/http/TestHtmlFilterSerializer.java
package com.ustadmobile.port.sharedse.impl.http;

import com.ustadmobile.port.sharedse.impl.http.EpubHtmlFilterSerializer;

import org.junit.Assert;
import org.junit.Test;
import org.xmlpull.v1.XmlPullParserException;

import java.io.IOException;
import java.io.InputStream;

/**
 * Created by mike on 11/4/17.
 */

public class TestHtmlFilterSerializer {

    @Test
    public void testSerializer() throws IOException, XmlPullParserException{
        EpubHtmlFilterSerializer serializer = new EpubHtmlFilterSerializer();
        InputStream in = getClass().getResourceAsStream("/com/ustadmobile/port/sharedse/epub-page.html");
        serializer.setIntput(in);
        serializer.setScriptSrcToAdd("/path/to/script");
        byte[] filteredInput = serializer.getOutput();
        String filteredStr = new String(filteredInput, "UTF-8");
        Assert.assertNotNull(filteredStr);
    }
}

File: sharedse/src/jvmTest/kotlin/com/ustadmobile/sharedse/impl/http/TestHtmlFilterSerializer.kt
package com.ustadmobile.sharedse.impl.http

import com.ustadmobile.port.sharedse.impl.http.EpubHtmlFilterSerializer
import org.junit.Assert
import org.junit.Before
import org.junit.Test
import org.kodein.di.*
import org.kxml2.io.KXmlParser
import org.kxml2.io.KXmlSerializer
import org.xmlpull.v1.XmlPullParser
import org.xmlpull.v1.XmlPullParserException
import org.xmlpull.v1.XmlPullParserFactory
import org.xmlpull.v1.XmlSerializer
import java.io.IOException
import java.nio.charset.Charset

/**
 * Created by mike on 11/4/17.
 */

class TestHtmlFilterSerializer {

    private lateinit var di: DI

    private lateinit var serializer: EpubHtmlFilterSerializer

    @Before
    fun setup() {
        di = DI {
            bind<XmlPullParserFactory>() with singleton {
                XmlPullParserFactory.newInstance().also {
                    it.isNamespaceAware = true
                }
            }

            bind<XmlPullParser>() with provider {
                instance<XmlPullParserFactory>().newPullParser()
            }

            bind<XmlSerializer>() with provider {
                KXmlSerializer()
            }
        }

        serializer = EpubHtmlFilterSerializer(di)
    }

    @Test
    fun givenXhtmlWithEntities_whenFiltered_thenShouldParseSuccessfully() {
        val `in` = javaClass.getResourceAsStream("/com/ustadmobile/port/sharedse/epub-page2.xhtml")
        serializer.setIntput(`in`)
        serializer.scriptSrcToAdd = "/path/to/script"
        val filteredInput = serializer.output
        val filteredStr = String(filteredInput, Charset.defaultCharset())
        Assert.assertNotNull(filteredStr)
    }

    @Test
    fun givenValidXhtmlPage_whenFiltered_thenShouldParseSuccessfully() {
        val `in` = javaClass.getResourceAsStream("/com/ustadmobile/port/sharedse/epub-page2.xhtml")
        serializer.setIntput(`in`)
        serializer.scriptSrcToAdd = "/path/to/script"
        val filteredInput = serializer.output
        val filteredStr = String(filteredInput, Charset.defaultCharset())
        Assert.assertNotNull(filteredStr)
    }
}
